kube-core/src/version.rs

```diff
@@ -71,7 +131,7 @@ impl Version {
 
 impl Ord for Version {
     fn cmp(&self, other: &Self) -> std::cmp::Ordering {
```

Comment:
I think this is actually the key question. Because it feels like the biggest disagreement here is how semantic this `Ord` impl really is.

It is currently explicitly modelling "**version priority**", in that a version with a higher priority will be picked before a version with lower priority by `kubectl`.

The [key document](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/#version-priority) explicitly talks about it like a sort order:

> "The algorithm used for sorting the versions is designed to sort versions in the same way that the Kubernetes project sorts Kubernetes versions".

so personally, I think it makes sense to co-opt `Ord` for this, because kubectl explicitly sorts version using this order.

Reply:
https://github.com/kube-rs/kube-rs/pull/767

Reply:
Have merged in the alternate proposal in the state we left it in. We have two `impl Ord`'s now, but `Version::priority` exposes version priority, so that should be pretty clear now.